### PR TITLE
New code formatting settings

### PR DESCRIPTION
**Pull request by hudochenkov**
> Which issue, if any, is this issue related to?

https://github.com/stylelint/stylelint/issues/4246

> Is there anything in the PR that needs further explanation?

Here is configuration only. I run Prettier with this settings and here the output: https://github.com/stylelint/stylelint/tree/new-code-formatting-output

Compare: https://github.com/stylelint/stylelint/compare/new-formatting-configuration...new-code-formatting-output

### CODE REVIEW

**Comment by vankop** [comment]
Снимок экрана 2019-09-06 в 12 03 47

vanilla github 😔

**Comment by hudochenkov** [comment]
Yeah. GitHub still didn't add CSS to show tabs smaller. I listed solutions in https://github.com/stylelint/stylelint/issues/4246. I use [Tab Size on GitHub](https://github.com/sindresorhus/tab-size-on-github).

**Comment by hudochenkov** [comment]
Pretest is failing because files are not formatted. This is suppose to happen.

File: .eslintignore [hunk 1]
```diff
@@ -2,3 +2,6 @@
 flow-typed
 lib/vendor
 lib/utils/parseCalcExpression/parser.js
+
+# Unignore config files like .prettierrc.js, because they're ignored by default
+!.*rc.js
```
Comment (ntwb) [inline .eslintignore]: Have you got a reference for this? I would be interested to have a read on why ESLint chooses to ignore these by default 🤔

**Comment by ntwb** [comment]
It looks like the _prettier_ fixes need to be completed before the CI job will pass.

Happy for you to override that and merge as is @hudochenkov and follow up the actual changes in a second pull request, instead of pushing the changes to this branch, up to you Aleks

File: .eslintignore (hunk 1 above)
Comment (hudochenkov) [reply .eslintignore]: Nope, and I can't find default ignore list on ESLint website as well. I faced this few years ago for the first time. Found solution, and now copied this solution to stylelint.

Comment (hudochenkov) [reply .eslintignore]: https://github.com/eslint/eslint/blob/7621f5d2aa7d87e798b75ca47d6889c280597e99/lib/cli-engine/ignored-paths.js#L34 They ignore dotfiles by default.

**Comment by jeddy3** [comment]
What will be the impact on:

- open PRs e.g. will they be littered with conflicts?
- diff history?

In https://github.com/stylelint/stylelint/issues/3112 we postponed using lebab to add modern syntax because it "messes with understanding the diff and history" (https://github.com/stylelint/stylelint/issues/3112#issuecomment-358177571)

**Comment by hudochenkov** [comment]
> open PRs e.g. will they be littered with conflicts?

They will. Luckily we have only few unmerged PRs, that could be merged soon. The rest of PRs are stale, which should not hold stylelint progress.

> diff history?
> In #3112 we postponed using lebab to add modern syntax because it "messes with understanding the diff and history" (#3112 (comment))

For every JS file there will be commit with reformatting. With blame function in Github it's not hard to see the past state of a file.

**Comment by jeddy3** [comment]
Thanks for the answers. I'm happy to see this PR go in if the changes help yourself and other maintainers who touch the code often work more easily (as I haven't had time to touch the code in a while).

**Comment by hudochenkov** [comment]
@stylelint/contributors I would like to see more reviews, please :)

File: .editorconfig [hunk 2]
```diff
@@ -4,7 +4,6 @@ root = true
 [*]
 charset = utf-8
 end_of_line = lf
-indent_size = 2
-indent_style = space
+indent_style = tab
```
Comment (Maxim-Mazurok) [inline .editorconfig]: This one seems pretty aggressive :)
Tabs vs spaces is a holy war, let's live and code in peace :)

Comment (Maxim-Mazurok) [reply .editorconfig]: Sorry, I went through your arguments. Seems ok to me. Dying to see a good visual programming IDE, that would solve indentation struggle once and for all :)

File: .prettierrc.js [hunk 3]
```diff
@@ -0,0 +1,10 @@
+'use strict';
+
+// https://prettier.io/docs/en/options.html
+module.exports = {
+	arrowParens: 'always',
+	printWidth: 100,
+	singleQuote: true,
+	trailingComma: 'es5',
```
Comment (vkrol) [inline .prettierrc.js]: What about the "all" option? The minimal supported Node.js version is `8.7.0`, so we can safely use it.

Comment (hudochenkov) [reply .prettierrc.js]: `all` is fine by me.

**Comment by vankop** [comment]
👍

File: .prettierrc.js (hunk 3 above)
Comment (vkrol) [inline .prettierrc.js]:
```suggestion
	trailingComma: 'all',
```

**Comment by m-allanson** [comment]
I don't have strong opinions on this, happy to roll with whatever folks deem sensible. 

I think having to install a browser extension to fix GitHub's [default tab styling](https://github.com/stylelint/stylelint/pull/4259#issuecomment-528775693) is something casual contributors are unlikely to do. And as a separate example, it seems like viewing diffs on a mobile phone will be much more difficult after this change. 

That said, if it makes life easier for regular contributors, then that's probably a worthwhile tradeoff. 

👍 Go wild :)

**Comment by hudochenkov** [comment]
Thank you, all!

**Pull request merged**